torch/_inductor/fx_passes/quantization.py

```diff
@@ -1209,6 +1231,33 @@ def _register_woq_mm_int8_pattern4():
     _register_woq_lowering(_woq_pattern, aten._weight_int8pack_mm.default, aten.reshape)
 
 
+def _register_woq_mm_int4_lowering():
+    computation_op = aten._weight_int4pack_mm_for_cpu.default
+    pattern = CallFunction(
+        computation_op,
+        KeywordArg("x"),
+        KeywordArg("weight"),
+        KeywordArg("qGroupSize"),
+        KeywordArg("qScaleAndZeros"),
+    )
+
+    @register_lowering_pattern(
+        pattern,
+        extra_check=_is_valid_woq_optimization_pattern(is_int4=True),
+    )
+    def woq_int4(match: Match, *args, **kwargs):
+        x = kwargs["x"]
+        weight = kwargs["weight"]
+        qGroupSize = kwargs["qGroupSize"]
+        qScaleAndZeros = kwargs["qScaleAndZeros"]
+        counters["inductor"]["woq_matcher_count"] += 1
+        counters["inductor"]["woq_matcher_nodes"] += len(match.nodes)
+
+        return L[computation_op](x, weight, qGroupSize, qScaleAndZeros)
+
+    return woq_int4
+
+
```

Comment:
@Xia-Weiwen, can you please explain the motivation behind this code? It seems to be redundant (not required) because your changes in `torch/_inductor/quantized_lowerings.py` can be used without the changes you're making to `torch/_inductor/fx_passes/quantization.py`.

The corresponding changes in `torch/_inductor/fx_passes/quantization.py` were necessary for int8 WoQ GEMM because a decomposed pattern could be matched to it, and that's why @Valentine233 had added such a pass. However, since you're using `aten._weight_int4pack_mm_for_cpu` for int4 WoQ GEMM here directly, the changes you're proposing to `torch/_inductor/fx_passes/quantization.py` are redundant.

Thanks!

Reply:
Thanks. It's not needed. It's removed.